database/postgresDB.go

```diff
@@ -0,0 +1,22 @@
+package database
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+func NewPostgreSQLConnection(uri string, connections int32) (*pgxpool.Pool, error) {
+	config, err := pgxpool.ParseConfig(uri)
+	if err != nil {
+		return nil, err
+	}
+	config.MaxConns = connections
+	//create a new connection pool
+	pool, err := pgxpool.ConnectConfig(context.Background(), config)
+	if err != nil {
+		return nil, err
+	}
+	PostgreDB = pool
+	return pool, nil
```

Comment:
Also, since you're assigning the connection to the global connection variable...there's no need to return it again. Just return error only. So, update your function to only return error. 

If you are returning the connection, a user might end up reassigning the returned connection to another variable which means we end up with two references to the same connection pool.